### PR TITLE
Improve how we handle the MDC values so that we don't clear them too …

### PR DESCRIPTION
**Pull request by ekirkland**
Improve how we handle the MDC values so that we don't clear them too early and lose important information before things have been logged.

Doing the MDC stuff in the scope thing doesn't work, everyone has to remember to opt into it, and we clear it before we've logged some important info, making searching difficult if you expect all the relevant messages to have the correct message id field.  It's important that we can rely on this data so that we don't waste time when debugging.

This works, please suggest an alternate location for this logic if you don't want the calls directly in the subscriber.

### CODE REVIEW

**Comment by ekirkland** [comment]
@jd2wilson - here's what I was talking about.

**Comment by ekirkland** [comment]
@SpencerHaan - I'm reviving this one.  Let me know what version number you'd want for this and if you have any feedback.  We need to get this in to help with some of the logging initiatives we want to do.  If you want to improve it further to remove slf4j completely then we could do that later on but for now I'd like to get this in place.

File: pom.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>io.axonif</groupId>
   <artifactId>queuebacca</artifactId>
-  <version>0.2.2</version>
+  <version>0.2.3</version>
```
Comment (SpencerHaan) [inline pom.xml]: Don't worry about upgrading the version, I usually do it in a commit prior to putting it into the master branch.

File: queuebacca-core/src/main/java/io/axonif/queuebacca/Logger.java [hunk 2]
```diff
@@ -55,6 +57,15 @@ public void error(String format, Object... arguments) {
 		log(format, arguments, logger::error);
 	}
 
+	public void initalizeContext(Context context) {
```
Comment (SpencerHaan) [inline queuebacca-core/src/main/java/io/axonif/queuebacca/Logger.java]: `setContext()` I think makes more sense. Initialize seems to be more like an upfront, one time thing.

**Comment by ekirkland** [comment]
I put the default log level back to error here.  It works well when I override the behaviour in our application.  How does this look now?

**Pull request merged**